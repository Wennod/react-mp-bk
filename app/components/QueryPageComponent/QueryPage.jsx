import React from 'react';
import ResultsContainer from './ResultsComponent/ResultsContainer';
import Header from './HeaderComponent/Header';

class QueryPage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            queryResults: {},
            queryValue: '',
            queryParam: 'title'
        };
    }

    getMovies(event) {
        event.preventDefault();
        fetch(`http://react-cdp-api.herokuapp.com/movies?search=${this.state.queryValue}&searchBy=${this.state.queryParam}`).then((result) => {
            return result.json();
        }).then((result) => {
            this.setState(() => {
                return { queryResults: result }; 
            });
        }, (err) => {
            throw new Error(err);
        });
    }

    setQueryValue(input) {
        this.setState(() => {
            return { queryValue: input.value };
        });
    }

    setQueryParam(radio) {
        this.setState(() => {
            return { queryParam: radio.value };
        });
    }

    render() {
        return (
            <React.Fragment>
                <Header setQueryValue={this.setQueryValue.bind(this)} getMovies={this.getMovies.bind(this)} setQueryParam={this.setQueryParam.bind(this)} />
                <ResultsContainer data={this.state.queryResults.data} />
            </React.Fragment>
        );
    }
}

export default QueryPage;