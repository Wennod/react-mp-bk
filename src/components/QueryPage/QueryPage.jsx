import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import Header from '../Header/Header';
import ResultsContainer from './ResultsContainer/ResultsContainer';
import Footer from '../Footer/Footer';

import style from './QueryPage.style.scss';
import * as queryPageActions from '../../actions/QueryPage.actions';

class QueryPage extends React.Component {
    constructor(props) {
        super(props);
    }

    getMovies(event) {
        event.preventDefault();
        fetch(this.getQueryString())
            .then(result => result.json())
            .then((result) => {
                    this.setState(() => ({ queryResults: result }));
                },
                (err) => {
                    throw new Error(err);
                }
            );
    }

    render() {
        console.log(this.props);
        return (
            <div className="query-page">
                <Header getMovies={this.props.queryPageActions.fetchMovies} />
                <ResultsContainer />
                <Footer />
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        queryPage: state.query,
        search: state.search
    };
};

function mapDispatchToProps(dispatch) {
    return {
        queryPageActions: bindActionCreators(queryPageActions, dispatch)
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(QueryPage)
