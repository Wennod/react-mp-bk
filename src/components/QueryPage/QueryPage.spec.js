import React from 'react';
import QueryPage from './QueryPage';
import fetch from 'isomorphic-fetch';

describe('QueryPage snapshot', () => {
    it('should match snapshot', () => {
        const tree = mount(<QueryPage />);
        expect(tree).toMatchSnapshot();
    });
});

describe('QueryPage test', () => {
    beforeEach(() => {
        window.fetch = jest.fn().mockImplementation(() => Promise.resolve('value'));
    });
    
    it('should trigget fetch', () => {
        const tree = mount(<QueryPage />);
        const button = tree.find('.search-button');
        button.simulate('submit');
        expect(window.fetch).toBeCalled();
    });

    it('should change param to title', () => {
        const tree = mount(<QueryPage />);        
        const radio = tree.find('#title');
        radio.simulate('click');
        expect(tree.state().queryParam).toEqual('title');
    });

    it('should change param to genres', () => {
        const tree = mount(<QueryPage />);        
        const radio = tree.find('#genre');
        radio.simulate('click');
        expect(tree.state().queryParam).toEqual('genres');
    });

    it('should change search value', () => {
        const tree = mount(<QueryPage />);     
        const input = tree.find('.search-form__input');
        console.log(input);
    });
});