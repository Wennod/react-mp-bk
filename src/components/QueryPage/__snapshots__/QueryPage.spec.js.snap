// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryPage snapshot should match snapshot 1`] = `
<QueryPage>
  <div
    className="query-page"
  >
    <Header
      getMovies={[Function]}
      setQueryParam={[Function]}
      setQueryValue={[Function]}
    >
      <div
        className="header"
      >
        <div
          className="header__logo-wraper"
        >
          <img
            alt="logo"
            className="header__logo"
            src="http://via.placeholder.com/100x25"
          />
        </div>
        <div
          className="header__search-wraper"
        >
          <SearchComponentContainer
            className="header__search-component"
            getMovies={[Function]}
            setQueryParam={[Function]}
            setQueryValue={[Function]}
          >
            <SearchForm
              getMovies={[Function]}
              setQueryParam={[Function]}
              setQueryValue={[Function]}
            >
              <form
                className="search-form"
                onSubmit={[Function]}
              >
                <h2>
                  Find your movies
                </h2>
                <input
                  className="search-form__input"
                  id="fieldId"
                  name="search"
                  onChange={[Function]}
                  type="text"
                />
                <div
                  className="search-form__group"
                >
                  <div
                    className="search-form__filter"
                  >
                    <p
                      className="search-form__filter-title"
                    >
                      Search By
                    </p>
                    <input
                      className="search-form__filter-toggler search-form__filter-toggler--t"
                      defaultChecked={true}
                      id="title"
                      name="searchCriteria"
                      onClick={[Function]}
                      type="radio"
                      value="title"
                    />
                    <input
                      className="search-form__filter-toggler search-form__filter-toggler-g"
                      id="genre"
                      name="searchCriteria"
                      onClick={[Function]}
                      type="radio"
                      value="genres"
                    />
                  </div>
                  <input
                    className="search-button"
                    type="submit"
                    value="SEARCH"
                  />
                </div>
              </form>
            </SearchForm>
          </SearchComponentContainer>
        </div>
      </div>
    </Header>
    <ResultsContainer>
      <div
        className="results results--nodata"
      >
        <div
          className="results__filter-container"
        >
          <ResultsPane
            isData={false}
          >
            <div
              className="result-filter result-filter--nodata"
            />
          </ResultsPane>
        </div>
        <div
          className="results__list-container"
        >
          <ResultList>
            <div
              className="list"
            >
              <h2
                className="results__error"
              >
                No results
              </h2>
            </div>
          </ResultList>
        </div>
      </div>
    </ResultsContainer>
    <Footer>
      <footer
        className="footer"
      >
        <img
          alt="logo"
          className="footer__logo"
          src="http://via.placeholder.com/100x25"
        />
      </footer>
    </Footer>
  </div>
</QueryPage>
`;
